Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/NMDataServiceTest.cs

```diff
@@ -137,52 +137,134 @@ public void WhenGetDailyBatchDetailsFilesIsCalled_ThenShouldExecuteCatch()
+        [Test]
+        public async Task WhenGetAllYearWeekIsCalledwithValidToken_ThenShouldReturnYearWeekList()
+        {
+            int ExpectedRecordCount = 3;
+            A.CallTo(() => _fakeAuthFssTokenProvider.GenerateADAccessToken(A<string>.Ignored));
+
+            IResult<BatchAttributesSearchResponse> res = SetAttributeSearchResult();
+            A.CallTo(() => _fakefileShareService.FssSearchAttributeAsync("", CorrelationId)).Returns(res);
 
-            List<KeyValuePair<string, string>> result = _nMDataService.GetAllWeeksOfYear(year, CorrelationId);
+            List<YearWeekModel> result = await _nMDataService.GetAllYearWeek(CorrelationId);
 
-            Assert.AreEqual(totalWeeks + 1, result.Count);
+            Assert.IsNotEmpty(result);
+            Assert.AreEqual(result.Count, ExpectedRecordCount);
         }
 
         [Test]
-        public void WhenGetAllYearsIsCalled_ThenShouldReturn4Records()
+        public async Task WhenGetWeeklyFilesResponseModelsAsyncIsCalled_ThenShouldReturnsShowWeeklyFilesResponseModelCount()
         {
-            const int yearsCount = 4;
-            List<KeyValuePair<string, string>> result = _nMDataService.GetAllYears(CorrelationId);
-            Assert.AreEqual(yearsCount, result.Count);
+            const int year = 2022, week = 15, expectedShowFilesResponseModelRecordCount = 4, expectedYearAndWeekRecordCount = 3;
+
+            A.CallTo(() => _fakeAuthFssTokenProvider.GenerateADAccessToken(A<string>.Ignored));
+
+            Result<BatchSearchResponse> searchResult = SetSearchResultForDaily();
+            A.CallTo(() => _fakefileShareService.FssBatchSearchAsync(A<string>.Ignored, A<string>.Ignored, CorrelationId)).Returns(searchResult);
+
+            IResult<BatchAttributesSearchResponse> res = SetAttributeSearchResult();
+            A.CallTo(() => _fakefileShareService.FssSearchAttributeAsync("", CorrelationId)).Returns(res);
+
+            ShowWeeklyFilesResponseModel showWeeklyFilesResponseModel = await _nMDataService.GetWeeklyFilesResponseModelsAsync(year, week, CorrelationId);
+
+            Assert.AreEqual(expectedYearAndWeekRecordCount, showWeeklyFilesResponseModel.YearAndWeek.Count);
+            Assert.AreEqual(expectedShowFilesResponseModelRecordCount, showWeeklyFilesResponseModel.ShowFilesResponseModel.Count);
         }
 
         [Test]
-        public void WhenGetAllYearsIsCalled_ThenShouldCheckMinYear()
+        public async Task WhenGetWeeklyFilesResponseModelsAsyncWithZeroIsCalled_ThenShouldReturnsShowWeeklyFilesResponseModelCount()
         {
-            int minYear = DateTime.Now.Year - 2;
-            List<KeyValuePair<string, string>> result = _nMDataService.GetAllYears(CorrelationId);
-            Assert.AreEqual(minYear.ToString(), result.LastOrDefault().Value);
+            const int year = 0, week = 0, expectedShowFilesResponseModelRecordCount = 4, expectedYearAndWeekRecordCount = 3;
```

Comment:
Please split these declarations onto separate lines:
```suggestion
            const int year = 0;
            const int week = 0;
            const int expectedShowFilesResponseModelRecordCount = 4;
            const int expectedYearAndWeekRecordCount = 3;
```

Reply:
Implemented suggested change